### ISSUE TITLE
CLICON_CONFIGDIR with external subsystems causes endless looping

### ISSUE THREAD

**Issue by pprindeville** [clicon/clixon#439]
In the XML parsing in `parse_configfile_one()` we have the [following loop](https://github.com/clicon/clixon/blob/master/lib/src/clixon_options.c#L329-L342).

The problem with the loop is that you can't "ignore" the nodes by just doing a `continue`.  They need to be deleted from `xe` so that the next call to examine the head of the vector:

```
while ((xec = xml_child_i_type(xe, 0, CX_ELMNT)) != NULL) {
```

doesn't find them still there.  We need to do a `xml_child_rm()` to remove them from `xe` before continuing the loop.

**Comment by pprindeville** [comment]
Fixed in PR #440.

**Comment by pprindeville** [comment]
Fixed w/ the merge of PR #440.
